javaslang/src/main/java/javaslang/collection/JavaConverters.java

```diff
@@ -0,0 +1,563 @@
+        }
+
+        @Override
+        public int lastIndexOf(Object obj) {
+            return Seq.<Object>narrow(delegate).lastIndexOf(obj);
+        }
+
+        @Override
+        public ListIterator<T> listIterator() {
+            return delegate.toJavaList().listIterator();
+        }
+
+        @Override
+        public ListIterator<T> listIterator(int index) {
+            return delegate.toJavaList().listIterator(index);
+        }
+
+        @Override
+        public T remove(int index) {
+            // may throw an IndexOutOfBoundsException accordingly to the j.u.List.remove(int)
+            return setDelegateAndGetPreviousElement(index, delegate.removeAt(index));
+        }
+
+        @Override
+        public boolean remove(Object o) {
+            // may throw a ClassCastException accordingly to the j.u.List.remove(Object)
+            @SuppressWarnings("unchecked") final T that = (T) o;
+            return setDelegateAndCheckChanged(delegate.remove(that));
+        }
+
+        @Override
+        public boolean removeAll(Collection<?> c) {
+            // may throw a ClassCastException accordingly to the j.u.List.removeAll(Collection)
+            @SuppressWarnings("unchecked") final Collection<T> that = (Collection<T>) c;
+            return setDelegateAndCheckChanged(delegate.removeAll(that));
+        }
+
+        @Override
+        public boolean retainAll(Collection<?> c) {
+            // may throw a ClassCastException accordingly to j.u.List.retainAll(Collection)
+            @SuppressWarnings("unchecked") final Collection<T> that = (Collection<T>) c;
+            return setDelegateAndCheckChanged(delegate.retainAll(that));
+        }
+
+        @Override
+        public T set(int index, T element) {
+            // may throw an IndexOutOfBoundsException accordingly to the j.u.List.set(int, T)
+            return setDelegateAndGetPreviousElement(index, delegate.update(index, element));
+        }
+
+        @Override
+        public int size() {
+            return delegate.size();
+        }
+
+        @Override
+        public List<T> subList(int fromIndex, int toIndex) {
+            // may throw IndexOutOfBoundsException accordingly to j.u.List.subList(int, int)
+            return delegate.subSequence(fromIndex, toIndex).asJava();
+        }
+
+        @Override
+        public Object[] toArray() {
+            return delegate.toJavaArray();
+        }
+
+        @Override
+        public <U> U[] toArray(U[] array) {
+            return delegate.toJavaList().toArray(array);
+        }
+
+        // -- Object.*
+
+        @Override
+        public boolean equals(Object o) {
+            return (o == this) || ((o instanceof java.util.List) && areEqual(delegate, (java.util.List<?>) o));
+        }
+
+        @Override
+        public int hashCode() {
+            return delegate.hashCode();
+        }
+
+        @Override
+        public String toString() {
+            return delegate.toString();
```

Comment:
@paplorinc I see the output of List.toString() as part of the interface spec. We should return the same as Java:

```java
        List<Integer> list = new ArrayList<>();
        println(list);
        list.add(1);
        list.add(2);
        println(list);

        println(Collections.emptyList());
        println(Collections.singleton(1));
```

Output:

```java
[]
[1, 2]
[]
[1]
```

Please change the toString() of the List view as I initially suggested:

```java
        @Override
        public String toString() {
            return delegate.mkString("[", ", ", "]");
        }
```

**Note:** The views for the other j.u. collections should also produce output identically to existing Java impls, e.g.

* *AsJavaList.toString() = ArrayList.toString()
* *AsJavaMap.toString() = HashMap.toString()
* *AsJavaSet.toString() = HashSet.toString()

Set:

```java
        Set<Integer> set = new HashSet<>();
        println(set);
        set.add(1);
        set.add(2);
        println(set);

// output:
[]
[1, 2]
class java.util.HashSet
```

Map:

```java
        Map<Integer, String> map = new HashMap<>();
        println(map);
        map.put(1, "a");
        map.put(2, "b");
        println(map);

// output:
{}
{1=a, 2=b}
```